src/dos/cdrom.h

```diff
@@ -354,18 +349,42 @@ class CDROM_Interface_Image final : public CDROM_Interface
 };
 
 #if defined (LINUX)
-class CDROM_Interface_Ioctl : public CDROM_Interface_SDL
-{
+class CDROM_Interface_Ioctl : public CDROM_Interface {
 public:
-	CDROM_Interface_Ioctl		();
-
-	bool	SetDevice		(const char* path, const int cd_number);
-	bool	GetUPC			(unsigned char& attr, char* upc);
-	bool	ReadSectors		(PhysPt buffer, const bool raw, const uint32_t sector, const uint16_t num);
-	bool	ReadSectorsHost	(void* buffer, bool raw, unsigned long sector, unsigned long num);
+	~CDROM_Interface_Ioctl();
+
+	bool SetDevice(const char* path, const int cd_number);
+	bool GetUPC(unsigned char& attr, char* upc);
+	bool GetAudioTracks(uint8_t& stTrack, uint8_t& end, TMSF& leadOut);
+	bool GetAudioTrackInfo(uint8_t track, TMSF& start, unsigned char& attr);
+	bool GetAudioSub(unsigned char& attr, unsigned char& track,
+	                 unsigned char& index, TMSF& relPos, TMSF& absPos);
+	bool GetAudioStatus(bool& playing, bool& pause);
+	bool GetMediaTrayStatus(bool& mediaPresent, bool& mediaChanged,
+	                        bool& trayOpen);
+	bool ReadSectors(PhysPt buffer, const bool raw, const uint32_t sector,
+	                 const uint16_t num);
+	bool ReadSectorsHost(void* buffer, bool raw, unsigned long sector,
+	                     unsigned long num);
+	bool PlayAudioSector(const uint32_t start, uint32_t len);
+	bool PauseAudio(bool resume);
+	bool StopAudio();
+	void ChannelControl(TCtrl ctrl);
+	bool LoadUnloadMedia(bool unload);
 
 private:
-	char	device_name[512];
+	void CDAudioCallback(uint16_t requested_frames);
+	void InitAudio();
+
+	int cdrom_fd                            = -1;
+	mixer_channel_t mixer_channel           = nullptr;
+	std::unique_ptr<int16_t[]> audio_buffer = nullptr;
```

Comment:
It would be better to use `std::vector`.

Reply:
I was considering that.  I'm never using any vector operations though and it's always a fixed size.  The buffer is only used to read in one second's worth of data from a read `ioctl` and then pass it on (a millisecond or so at a time) to `AddSamples_s16`.  Both of which take raw pointers.

I'll change it if you want.  I don't have strong feeling about it.  But all I would be doing is calling `.data()` on it to get to the raw buffer so I'm not really sure what the benefit is.

Reply:
Yeah, fair points but vector is just nicer and safer, and that would remove the need for the unique_ptr cruft and memset. Not that important for this specific use case, but it contributes to the overall consistency of the codebase. And in most cases there is no reason to really _prefer_ raw arrays over vector or even std:array.

Reply:
I changed this to use `std::vector`